dvc/repo/ls.py

```diff
@@ -44,34 +72,39 @@ def _ls_files_repo(target_path_info, recursive=None):
     tree = CleanTree(WorkingTree(target_path_info))
     try:
         for dirpath, dirnames, filenames in tree.walk(target_path_info):
-            files.extend(map(lambda f: PathInfo(dirpath, f), filenames))
+            files.extend(
+                map(lambda f: FSNode(PathInfo(dirpath, f)), filenames)
+            )
             if not recursive:
-                files.extend(map(lambda d: PathInfo(dirpath, d), dirnames))
+                files.extend(
+                    map(lambda d: FSNode(PathInfo(dirpath, d)), dirnames)
+                )
                 break
     except NotADirectoryError:
         if os.path.isfile(fspath(target_path_info)):
-            return [target_path_info]
+            return [FSNode(target_path_info)]
 
```

Comment:
Maybe let's wrap `files` in one place instead of wrapping every line in `FSNode`?